src/auth0-provider-with-history.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Auth0Provider } from '@auth0/auth0-react';
+import { useHistory } from 'react-router-dom';
+
+const Auth0providerWithHistory = ({ children }) => {
+    const history = useHistory();
+    const domain = process.env.REACT_APP_AUTHO_DOMAIN;
+    const clientId = process.env.REACT_APP_AUTHO_CLIENT_ID;
+
+    const onRedirectCallback = (appState) => history.push(appState?.returnTo);
+
+    return (
+        <Auth0Provider
+            domain={domain}
+            clientId={clientId}
+            redirectUri={window.location.origin}
```

Comment:
dont use window.location, use react-router

Reply:
useHistory or useRouter hook

Reply:
I have updated this to use .env to dynamically direct to the home page on logout. I was not able to get it to work properly with useHistory.